modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/service/impl/RedirectEntryLocalServiceImpl.java

```diff
@@ -88,9 +89,14 @@ public RedirectEntry addRedirectEntry(
 			boolean permanent, String sourceURL, ServiceContext serviceContext)
 		throws PortalException {
 
-		_validate(destinationURL, sourceURL);
+		String normalizedSourceURL =
+			_friendlyURLNormalizer.normalizeWithEncoding(sourceURL);
+
+		_validate(destinationURL, normalizedSourceURL);
```

Comment:
Q: Shouldn't we normalize the destinationURL as well?

Reply:
The problem with the destination is that a full URL so the normalization of the will be 

http://www.google.com > http-//www.google.com 

and we do not have a de-normalization method because the changes can be made in several characters that we will do not know what are the original ones.

Reply:
Well
I'm thinking on a side-step that is to detect if the destination is an URL of the system and normalize the last part. (the friendly url)

what do you think about it?

Reply:
We've had a conversation in slack about this, and we've decided to move forward with this implementation.